modules/registrar.go

```diff
@@ -186,6 +186,6 @@ func buildRemoteSources(cfg *remote.Details) (*Sources, error) {
 		GovSource:      remotegovsource.NewSource(source, govtypes.NewQueryClient(source.GrpcConn)),
 		MintSource:     remotemintsource.NewSource(source, minttypes.NewQueryClient(source.GrpcConn)),
 		SlashingSource: remoteslashingsource.NewSource(source, slashingtypes.NewQueryClient(source.GrpcConn)),
-		StakingSource:  remotestakingsource.NewSource(source, stakingtypes.NewQueryClient(source.GrpcConn)),
+		StakingSource:  remotestakingsource.NewSource(source, stakingtypes.NewQueryClient(source.GrpcConn), slashingtypes.NewQueryClient(source.GrpcConn)),
```

Comment:
Instead of having dependencies between sources, we should have dependencies between modules instead. So a better alternative would be to:
1. remove the `slashing.QueryClient` field from the `staking.Source` 
2. have a new method called `GetValidatorSigningInfo` inside the `staking.Module` interface
3. have a new field of type `SlashingModule` inside the `StakingModule`, and then call `slashingModule.GetValidatorSigningInfo` when it's required to know the tombstoned value of a validator